tests/test_utils/gcp_system_helpers.py

```diff
@@ -68,3 +77,92 @@ def provide_gcp_context(
     return provide_gcp_conn_and_credentials(
         key_file_path=key_file_path, scopes=scopes, project_id=project_id
     )
+
+
+class GoogleSystemTest(SystemTest):
+    @staticmethod
+    def _project_id():
+        return os.environ.get("GCP_PROJECT_ID")
+
+    @staticmethod
+    def _service_key():
+        return os.environ.get(CREDENTIALS)
+
+    @contextmanager
+    def authentication(self):
+        self.authenticate()
+        yield
+        self.revoke_authentication()
+
+    @classmethod
+    def authenticate(cls):
```

Comment:
Can you inline this method in `authentication` context.  This will force these methods to be called correctly.